cypress/integration/13-timeline/timeline.spec.ts

```diff
@@ -0,0 +1,141 @@
+import type { MatrixClient } from "matrix-js-sdk/src/client";
+import { SynapseInstance } from "../../plugins/synapsedocker";
+import { SettingLevel } from "../../../src/settings/SettingLevel";
+import Chainable = Cypress.Chainable;
+
+// The avatar size used in the timeline
+const AVATAR_SIZE = 30;
+// The resize method used in the timeline
+const AVATAR_RESIZE_METHOD = "crop";
+
+const ROOM_NAME = "Test room";
+const OLD_AVATAR = "avatar_image1";
+const NEW_AVATAR = "avatar_image2";
+const OLD_NAME = "Alan";
+const NEW_NAME = "Alan (away)";
+
+const getEventTilesWithBodies = (): Chainable<JQuery> => {
+    return cy.get(".mx_EventTile").filter((_i, e) => e.getElementsByClassName("mx_EventTile_body").length > 0);
+};
+
+const expectDisplayName = (e: JQuery<HTMLElement>, displayName: string): void => {
+    expect(e.find(".mx_DisambiguatedProfile_displayName").text()).to.equal(displayName);
+};
+
+const expectAvatar = (e: JQuery<HTMLElement>, avatarUrl: string): void => {
+    cy.getClient().then((cli: MatrixClient) => {
+        expect(e.find(".mx_BaseAvatar_image").attr("src")).to.equal(
+            // eslint-disable-next-line no-restricted-properties
+            cli.mxcUrlToHttp(avatarUrl, AVATAR_SIZE, AVATAR_SIZE, AVATAR_RESIZE_METHOD),
+        );
+    });
+};
+
+const sendEvent = (roomId: string): Chainable<ISendEventResponse> => {
+    return cy.sendEvent(
+        roomId,
+        null,
+        "m.room.message" as EventType,
+        MessageEvent.from("Message").serialize().content,
+    );
+};
+
+describe("Timeline", () => {
+    let synapse: SynapseInstance;
+
+    let roomId: string;
+
+    let oldAvatarUrl: string;
+    let newAvatarUrl: string;
+
+    describe("useOnlyCurrentProfiles", () => {
+        beforeEach(() => {
+            cy.startSynapse("default").then(data => {
+                synapse = data;
+                cy.initTestUser(synapse, OLD_NAME).then(() =>
+                    cy.window({ log: false }).then(() => {
+                        cy.createRoom({ name: ROOM_NAME }).then(_room1Id => {
+                            roomId = _room1Id;
+                        });
+                    }),
+                ).then(() => {
+                    cy.uploadContent(OLD_AVATAR).then((url) => {
+                        oldAvatarUrl = url;
+                        cy.setAvatarUrl(url);
+                    });
+                }).then(() => {
+                    cy.uploadContent(NEW_AVATAR).then((url) => {
+                        newAvatarUrl = url;
+                    });
+                });
+            });
+        });
+
+        afterEach(() => {
+            cy.stopSynapse(synapse);
+        });
+
+        it("should show historical profiles if disabled", () => {
+            cy.setSettingValue("useOnlyCurrentProfiles", null, SettingLevel.ACCOUNT, false);
+            sendEvent(roomId);
+            cy.setDisplayName("Alan (away)");
+            cy.setAvatarUrl(newAvatarUrl);
+            cy.wait(500);
```

Comment:
comment for why please (shouldn't be required)

it's often a good idea in cypress tests to drive the UI rather than make API calls so you get the natural benefits of waiting for certain things to happen